### DISCUSSION TITLE
Budibase slow? data provider infinite loading, browser freezing & crash. (BB cloud)

### DISCUSSION THREAD

**Discussion by MikeYates** [Budibase/budibase#11956] [Help] [answered]
Is anyone having the same issue as me? my budibase builder / frontend is freezing, any component using data provider just shows loading infinitely. This just happened 15 minutes ago.

I recently installed some plugins (which didnt work so I uninstalled it) dont know if it had anything to do with that?

**Comment by ConorWebb96** [comment]
Hey @MikeYates,

I haven't noticed any slowness or freezing myself with cloud. We did have a new release yesterday so it might have been a bit slower because of that. Are you still experiencing issues? Might it be related to the other discussion, as well as the issue you created?

Thanks in advance.

**Reply by MikeYates** [reply]
It may be due to the update, I do remember updating it and I started to have this issue. But im only experiencing it on some
Of my screes which consists a lot of data providers and a lot of complex onChange & Conditions. Im looking into it and trying to recreate the issue.

I tried it on cloud & self host, same result. So it might be something related with my app.

**Reply by MikeYates** [reply, marked as answer]
@melohagan 

I found what the problem was, in one of my screen I had about 6 data provider. All of them are identical, but only one of them which kept loading infinitely caused my tab to crash, deleted that component and it was resolved. oddly it doesnt happen in builder but when I opened it in published mode it freezes, then my builder also would freeze too. Could be the filtering I used or my screen was too complex.

**Reply by melohagan** [reply]
Nice, glad it was fixed. It may not have been the issue in your case, but it is worth noting that if you are using bindings in your Data provider filters, the data providers will automatically re-fetch if the binding value changes.

Perhaps you had a strange setup were a binding value was being continuously updated, but just a stab in the dark.

**Reply by MikeYates** [reply]
May not have been the filter actually, could be the component inside the data provider. I have an option picker in each data provider and each had 5 onchange that resets default value of the 5 text field default value with math formulas. Could be an error in the bindings intertwining the components.

Anyhow, thank you for always been so helpful!
